R/plots.R

```diff
@@ -125,6 +125,133 @@ plot_pca <- function(obj,
   p
 }
 
+
+#' Plot Loadings and Interpretations 
+#' @param obj a \code{sleuth} object
+#' @param use_filtered if TRUE, use filtered data. otherwise, use all data
+#' @param PC is the (string) name of which principal component to view genes contribution to that PC
+#' @param gene_count user input on how many genes to display, otherwise default is 5
+#' @return a ggplot object
+#' @export
+plot_loadings <- function(obj, 
+  use_filtered = TRUE,
+  PC = NULL, #this is a string of the PC actual name
+  gene_count = NULL,
+  bool = FALSE,
+  ...) {
+
+  mat <- NULL
+  if (use_filtered) {
+    mat <- spread_abundance_by(obj$obs_norm_filt, units)
+  } else {
+    mat <- spread_abundance_by(obj$obs_norm, units)
+  }
+  
+  pca_calc <- prcomp(mat, scale = bool)
+  #transpose
+  loadings <- t(pca_calc$rotation)
+
+  if (!is.null(gene)) {
+    loadings <- loadings[, gene]
+  } else {
+    title <- colnames(loadings)[1]
+    loadings <- loadings[, 1]
+  }
+
+  if (!is.null(pc_count)) {
+    loadings <- loadings[1:pc_count]
+  } else {
+    loadings <- loadings[1:5]
+  }
+  #absolute value
+  if (absolute) {
+    loadings <- abs(loadings)
+  }
+  #sort loadings vector to obtain highest contribution
+  loadings <- sort(loadings, decreasing = TRUE)
+  names <- names(loadings)
+
+  dat <- data.frame(pc = names, loadings = loadings)
+  dat$pc <- factor(dat$pc, levels = unique(dat$pc))
+
+  p <- ggplot(dat, aes(x = pc, y = loadings)) 
+  p <- p + geom_bar(stat = "identity")
+  p <- p + xlab("Principal Components") + ylab("Contribution")
+
+  if (!is.null(gene)) {
+    p <- p + ggtitle(gene)
+  } else {
+    p <- p + ggtitle(title)
+  }
+
+  p
+
+}
+
+#' Plot PC variances retained by percentage with option to compare specified PC
+#'
+#' @param obj a \code{sleuth} object
+#' @param use_filtered if TRUE, use filtered data. otherwise, use all data
+#' @param pca_number user input on how many PC to display, otherwise default is 5
+#' @param bool determines scaling
+#' @param (integer) PC_relative gives the option to compare subsequent principal
+#' components and their contributions
+#' @return a ggplot object
+#' @export
+plot_pc_variance <- function(obj, 
+  use_filtered = TRUE,
+  pca_number = NULL,
+  bool = FALSE,
+  PC_relative = NULL, #this is an integer
+  ...) {
+
+  mat <- NULL
+  if (use_filtered) {
+    mat <- spread_abundance_by(obj$obs_norm_filt, units)
+  } else {
+    mat <- spread_abundance_by(obj$obs_norm, units)
+  }
+
+  pca_calc <- prcomp(mat, scale = bool) #PCA calculations 
+
+  #computation
+  eigenvalues <- (pca_calc$sdev)^2  
+  var <- eigenvalues*100/sum(eigenvalues)
```

Comment:
`var` is already the name of a built in R function. do not overwrite it. also, use a more descriptive name